cpm

```diff
@@ -119,15 +119,15 @@ fcount() {
 # list files in a directory (similar to ls -l)
 flist() {
     if [ -d "$1" ]; then
-        cd "$1"
-        printf '%s\n' *
+        cd "$1" || return
+        printf '%s\n' ./*
```

Comment:
Looking slightly deeper (sorry, only woke up half an hour ago), this is only used for slackpkg package listing, so this will make it list packages like

```
./pkg1
./pkg2
./other-package
```
rather than

```
pkg1
pkg2
other-pkg
```

If anything, I'd do

```
cd "$1" &&
printf '%s\n' *
```

I don't think shellcheck's complaints about using `./` matter here, as printf doesn't really take arguments (so files named `-h` or something won't make a difference).

Other than that, _I think_ it looks good.

Reply:
`| sed 's|./||g'`

?

Reply:
No.  You're still holding on to shellcheck too much.  It's a useful tool but it shouldn't be the be-all-end-all.

```
cd "$1" &&
printf '%s\n' *
```

Reply:
> No. You're still holding on to shellcheck too much. It's a useful tool but it shouldn't be the be-all-end-all.
> 
> ```
> cd "$1" &&
> printf '%s\n' *
> ```

![image](https://user-images.githubusercontent.com/47485207/116992269-4ae52c80-acc5-11eb-8aca-842873ffe851.png)

Reply:
some versions of printf do take options

Reply:
Ok, now try it in the way it's used in the script - worst case scenario.

```
printf '%s\n' --help
```

And what shell is it?

Reply:
And what shell is it? bash

Reply:
![image](https://user-images.githubusercontent.com/40295114/116992606-a8a36400-ad19-11eb-97c6-55693a6a092a.png)

Reply:
ok, I understand your point

welp I guess its "untold standard" that the option switches in printf have to come first

Reply:
Actually, it's the told standard that printf shouldn't have operands.

But bash, of course feels the need to be special.

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/printf.html

Reply:
(the email reply didn't continue here and made a comment on the main thread :/)